cpp/tools/parquet/parquet_dump_footer.cc

```diff
@@ -0,0 +1,135 @@
+  out->append(reinterpret_cast<const char*>(&v), sizeof(v));
+}
+
+int DoIt(std::string in, bool scrub, bool json, std::string out) {
+  std::string path;
+  auto fs = ::arrow::fs::FileSystemFromUriOrPath(in, &path).ValueOrDie();
+  auto file = fs->OpenInputFile(path).ValueOrDie();
+  int64_t file_len = file->GetSize().ValueOrDie();
+  if (file_len < 8) {
+    std::cerr << "File too short: " << in << "\n";
+    return 3;
+  }
+  // First do an opportunistic read of up to 1 MiB to try and get the entire footer.
+  int64_t tail_len = std::min(file_len, int64_t{1} << 20);
+  std::string tail;
+  tail.resize(tail_len);
+  char* data = tail.data();
+  file->ReadAt(file_len - tail_len, tail_len, data).ValueOrDie();
+  if (auto magic = ReadLE32(data + tail_len - 4); magic != ReadLE32("PAR1")) {
+    std::cerr << "Not a Parquet file: " << in << "\n";
+    return 4;
+  }
+  uint32_t metadata_len = ReadLE32(data + tail_len - 8);
+  if (tail_len >= metadata_len + 8) {
+    // The footer is entirely in the initial read. Trim to size.
+    tail = tail.substr(tail_len - (metadata_len + 8));
+  } else {
+    // The footer is larger than the initial read, read again the exact size.
+    if (metadata_len > file_len) {
+      std::cerr << "File too short: " << in << "\n";
+      return 5;
+    }
+    tail_len = metadata_len + 8;
+    tail.resize(tail_len);
+    data = tail.data();
+    file->ReadAt(file_len - tail_len, tail_len, data).ValueOrDie();
+  }
+  auto md = FileMetaData::Make(tail.data(), &metadata_len);
+  std::string ser = md->SerializeUnencrypted(scrub, json);
+  if (!json) {
+    AppendLE32(static_cast<uint32_t>(ser.size()), &ser);
+    ser.append("PAR1", 4);
+  }
+  std::optional<std::fstream> fout;
+  if (!out.empty()) fout.emplace(out, std::ios::out);
+  std::ostream& os = fout ? *fout : std::cout;
+  if (!os.write(ser.data(), ser.size())) {
+    std::cerr << "Failed to write to output file: " << out << "\n";
+    return 6;
+  }
+
+  return 0;
+}
+}  // namespace
+}  // namespace parquet
+
+static int PrintHelp() {
+  std::cerr << R"(Usage: parquet-dump-footer
+  -h|--help    Print help and exit
+  --no-scrub   Do not scrub potentially confidential metadata
+  --debug      Output text represenation of footer for inspection
+  --in <uri>   Input file (required): must be an URI or an absolute local path
+  --out <path> Output file (optional, default stdout)
+
+  Dump the footer of a Parquet file to stdout or to a file, optionally with
+  potentially confidential metadata scrubbed.
+)";
+  return 1;
+}
+
+int main(int argc, char** argv) {
+  bool scrub = true;
+  bool json = false;
+  std::string in;
+  std::string out;
+  for (int i = 1; i < argc; i++) {
+    char* arg = argv[i];
+    if (!std::strcmp(arg, "-h") || !std::strcmp(arg, "--help")) {
+      return PrintHelp();
+    } else if (!std::strcmp(arg, "--no-scrub")) {
+      scrub = false;
+    } else if (!std::strcmp(arg, "--json")) {
+      json = true;
```

Comment:
Should we use `debug` instead of `json` here?

Reply:
https://github.com/apache/arrow/pull/43375